www/kadm08/sp/myAccount.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+session_start();
+require_once __DIR__ . '/lib/UserDB.php';
+require __DIR__ . '/userRequired.php';
+
+$success = false;
+
+$errorMessages = [];
+$userDB = new UserDB();
+
+if (isset($_POST['user_info'])) {
+    $name = htmlspecialchars($_POST['name']);
+    $surname = htmlspecialchars($_POST['surname']);
+    $phone = htmlspecialchars($_POST['phone']);
+
+
+    if (empty($errorMessages)) {
+        $change_client = $userDB->updateClient($_SESSION['user_id'], $name, $surname, $phone);
+
+        $success = true;
+    }
+} elseif (isset($_POST['email_change'])) {
+    $email = $_POST['email'];
+    $confirmEmail = $_POST['email_confirm'];
+
+    if ($email != $confirmEmail) {
+        array_push($errorMessages, 'Your emails do not match.');
+    }
+
+    if (!filter_var($_POST['email'], FILTER_VALIDATE_EMAIL)) {
+        array_push($errorMessages, 'Please enter valid email address');
+    }
+
+    $change_email = $userDB->updateEmail($_SESSION['user_id'], $email);
+
+    $success = true;
+} elseif (isset($_POST['password_change'])) {
+    $password = $_POST['password'];
+    $confirmPassword = $_POST['password_confirm'];
+
+    if ($password != $confirmPassword) {
+        array_push($errorMessages, 'Your passwords do not match.');
+    } else if (strlen($password) < 5) {
+        array_push($errorMessages, 'Your password must be at least 5 characters long.');
+    }
+
+    if (empty($errorMessages)) {
+        $hashPassword = password_hash($password, PASSWORD_DEFAULT);
+
+        if (empty($errorMessages)) {
+
+            $change_password = $userDB->updatePassword($_SESSION['user_id'], $password);
```

Comment:
podle me tady ma byt predano $hashPassword

Reply:
Ano, to byla hloupa chyba, nejak mi to uniklo. Kvuli tomu nefungovala ani zmena hesla, jak zminujete v tom hlavnim komentari. Opraveno.